### PR TITLE
fix: proper event patching, powercycle vs. poweron

### LINKED ISSUE #145: Power cycle fails when server is powered off
```
020-10-09T21:21:23.140Z        ERROR   controllers.Server      failed to power cycle   {"server": "/REDACTED", "error": "run /usr/sbin/ipmitool ipmitool -H REDACTED -U REDACTED -P REDACTED -I lanplus raw 0x00 0x02 0x02: Unable to send RAW command (channel=0x0 netfn=0x0 lun=0x0 cmd=0x2 rsp=0xd5): Command not supported in present state\n (exit status 1)"}
```

### PR DESCRIPTION
**Pull request by smira**
Fixes #145

This fixes event recording, updates `patchServerInUse` to avoid
duplicate updates, adds condition to avoid multiple power cycle events.

We might want to expand on conditions to report better state to the
user.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

**Review by rsmitty — APPROVED** [review]

**Review by Unix4ever — APPROVED** [review]

File: app/metal-controller-manager/api/v1alpha1/server_types.go [hunk 1]
```diff
@@ -100,12 +101,16 @@ type ServerSpec struct {
 	Accepted          bool                    `json:"accepted"`
 }
 
+// ConditionPowerCycle is used to control the powercycle flow.
```
Comment (andrewrynhard) [inline app/metal-controller-manager/api/v1alpha1/server_types.go:104]: Name is PowerCycle (capital C) and then function on event log is Powercycle (lowercase c). We should be consistent. PowerCycle has my vote, but not a blocker for me.

**Review by andrewrynhard — DISMISSED** [review]

File: app/metal-controller-manager/api/v1alpha1/server_types.go (hunk 1 above)
Comment (smira) [reply app/metal-controller-manager/api/v1alpha1/server_types.go:104]: sure thing, let me update, I wasn't sure what looks better

**Review by andrewrynhard — DISMISSED** [review]

**Review by andrewrynhard — APPROVED** [review]

**Pull request merged**